documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+```
+
+
+### Client Stop
+
+In the v1 SDK, the `end` method in the `device` class disconnects the client. The force parameter determines whether to close
+the connection immediately, or wait for in-flight messages to be sent.
+
+In the v2 SDK, an MQTT5 client can stop a session by calling the [stop](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/classes/mqtt5.Mqtt5Client.html#stop)
+method. You can provide an optional [DisconnectPacket](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/interfaces/mqtt5.DisconnectPacket.html)
+parameter. A closed client can be started again by calling [start](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/classes/mqtt5.Mqtt5Client.html#start).
+
+#### Example of disconnecting a client in the v1 SDK
+
+```typescript
+device.end();
+```
+
+#### Example of disconnecting a client in the v2 SDK
+
+```typescript
+client.stop();
+```
+
+
+### Client Shutdown
+
+The v1 SDK automatically cleans resources on shutdown.
+
+In the v2 SDK, when an MQTT5 client is no longer required, your program **must** close it explicitly via a [close](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/classes/mqtt5.Mqtt5Client.html#close) call.
+
+#### Example of closing a client in the v2 SDK
+
+```typescript
+// Shutdown and clean up.
+const stopped = once(client, Mqtt5Client.STOPPED);
+client.stop();
+await stopped;
+
+// Release any native resources associated with the client.
+client.close();
+```
+
+
+### Reconnects
+
+In the v1 SDK, the `reconnectPeriod` option enables reconnection. The default value is 1000 ms which means it will try to
+reconnect one second after losing the connection.
+
+The v2 SDK attempts to reconnect automatically until connection succeeds or `client.stop()` is called. The reconnection parameters,
+such as min/max delays and [jitter modes](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/iot.AwsIotMqtt5ClientConfigBuilder.html#withRetryJitterMode),
+are configurable through [AwsIotMqtt5ClientConfigBuilder](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/iot.AwsIotMqtt5ClientConfigBuilder.html).
+
+
+#### Example of tweaking reconnection settings in the v1 SDK
+
+```typescript
+var device = deviceModule({
+    certPath: "<certificate file>",
+    // ...
+    reconnectPeriod: 1000,          // ms
+    minimumConnectionTimeMs: 2500,  // ms
+    maximumReconnectTimeMs: 10000   // ms
+});
+```
+
+#### Example of tweaking reconnection settings in the v2 SDK
+
+```typescript
+configBbuilder.withMinReconnectDelayMs(2500);
+configBbuilder.withMaxReconnectDelayMs(10000);
+configBbuilder.withRetryJitterMode(mqtt5.RetryJitterMode.Full);
+
+let config : mqtt5.Mqtt5ClientConfig = configBuilder.build();
+let client : mqtt5.Mqtt5Client = new mqtt5.Mqtt5Client(config);
+```
+
+
+### Offline Operations Queue
+
+The v1 SDK provides a set of options to configure behavior when a client is offline. By default, the number of queued messages
+are not limited.
```

Comment:
Trivial: As they're not limited, we might want to mention that this can lead to memory saturation. It can also tie to a note below